File: backend/domain/model/src/main/java/co/com/sofka/model/rutaaprendizaje/RutaAprendizaje.java
package co.com.sofka.model.rutaaprendizaje;

import lombok.*;

@Data
@Builder(toBuilder = true)
@NoArgsConstructor
@AllArgsConstructor
public class RutaAprendizaje {

    private String id;
    private String nombre;
    private String descripcion;
    private Integer nivel;
    //private Curso curso;
    private String prerrequisito;

    @Override
    public String toString() {
        return "RutaAprendizaje{" +
                "id='" + id + '\'' +
                ", nombre='" + nombre + '\'' +
                ", descripcion='" + descripcion + '\'' +
                ", nivel=" + nivel +
                //", curso=" + curso +
                ", prerrequisito='" + prerrequisito + '\'' +
                '}';
    }
}

File: frontend/src/app/models/ruta-aprendizaje.ts
export class RutaAprendizaje {
  id?: string;
  nombre: string;
  descripcion: string;
  nivel: number;
  //curso: Curso;
  prerrequisito: string;

  constructor(nombre: string, descripcion: string, nivel: number, prerrequisito: string) {
    this.nombre = nombre;
    this.descripcion = descripcion;
    this.nivel = nivel;
    this.prerrequisito = prerrequisito;
  }
}
